pulseapi/profiles/models.py

```diff
@@ -1,19 +1,117 @@
 from django.db import models
+from django.utils.html import format_html
+
+
+def entry_thumbnail_path(instance, filename):
+    return 'images/user-avatars/{timestamp}{ext}'.format(
+        timestamp=str(timezone.now()),
+        ext=os.path.splitext(filename)[1]
+    )
+
+
+# We want to capture the user or group's location, but we don't
+# want to do so in too fine a detail, so right now we mostly
+# care about city, and country that city is in.
+#
+class Location(models.Model):
+    """
+    THIS COMES WITH ASSUMPTION-PROBLEMS:
+
+    - A country might have two or more cities with the same name, but
+      in different administative regions of that country.
+
+    - Some cities lie in to countries. I know, that's crazy, but some do.
+
+    - Some people live or work or associate with more than one location,
+      especially if they live near an easily crossed border such as in
+      many European countries.
+
+    Making this do the right thing is going to take more work than any
+    initial model implementation can achieve.
+    """
+    city = models.CharField(max_length=500, blank=False, null=True)
```

Comment:
one thing to note about setting this to `False` is that for existing entries, once you open it with the intent of changing another property, it will fail to save until you also add a value for this field.